src/cascadia/TerminalApp/TitlebarControl.cpp

```diff
@@ -0,0 +1,100 @@
+﻿//
+// TitlebarControl.xaml.cpp
+// Implementation of the TitlebarControl class
+//
+
+#include "pch.h"
+
+#include "TitlebarControl.h"
+
+#include "TitlebarControl.g.cpp"
+
+namespace winrt::TerminalApp::implementation
+{
+    TitlebarControl::TitlebarControl()
+    {
+        const winrt::Windows::Foundation::Uri resourceLocator{ L"ms-appx:///TitlebarControl.xaml" };
+        winrt::Windows::UI::Xaml::Application::LoadComponent(*this, resourceLocator, winrt::Windows::UI::Xaml::Controls::Primitives::ComponentResourceLocation::Nested);
+
+        // Register our event handlers on the MMC buttons.
+        MinMaxCloseControl().MinimizeButton().Click({ this, &TitlebarControl::Minimize_Click });
+        MinMaxCloseControl().MaximizeButton().Click({ this, &TitlebarControl::Maximize_Click });
+        MinMaxCloseControl().CloseButton().Click({ this, &TitlebarControl::Close_Click });
+    }
+
+    Windows::UI::Xaml::UIElement TitlebarControl::Content()
+    {
+        return ContentRoot().Children().Size() > 0 ? ContentRoot().Children().GetAt(0) : nullptr;

```

Comment:
this is technically something you should use a control template and a template binding for.

Reply:
so that people can do `<TitleBar>Whatever</TitleBar>` and it will put Whatever in the Content

Reply:
That's certainly the direction I was going with this, but I wanted to try and split that into a different task - this one's already getting hefty. I can file a follow-up issue, yea?